### PR TITLE
add CUDA-7.5.18.eb

### PR DESCRIPTION
**Pull request by molden**
depends on PR https://github.com/hpcugent/easybuild-easyblocks/pull/708

### CODE REVIEW

File: easybuild/easyconfigs/c/CUDA/CUDA-7.5.18.eb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+homepage = 'https://developer.nvidia.com/cuda-toolkit'
+description = """CUDA (formerly Compute Unified Device Architecture) is a parallel
+ computing platform and programming model created by NVIDIA and implemented by the
+ graphics processing units (GPUs) that they produce. CUDA gives developers access
+ to the virtual instruction set and memory of the parallel computational elements in CUDA GPUs."""
+
+toolchain = {'name': 'dummy', 'version': 'dummy'}
+
+source_urls = ['http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/']
```
Comment (boegel) [inline easybuild/easyconfigs/c/CUDA/CUDA-7.5.18.eb]: please use `%(version_major_minor)s` rather than hardcoding `7.5`

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 8 out of 8 (8 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/b3b42f8905dd8c63ac93 for a full test report.

**Comment by boegel** [comment]
Tested on top of https://github.com/hpcugent/easybuild-easyblocks/pull/708, works great, so good to go.

Thanks @molden!

**Comment by thiell** [comment]
Hi,

I would like to report that for me if failed in HierarchicalMNS mode due to a weird hardcoded check in  det_modpath_extensions() method. I achieved to get it installed with the following patch, althrough I'm not sure it won't break anything else:

```diff
--- tools/module_naming_scheme/hierarchical_mns.py.orig	2015-10-13 19:12:52.299841778 -0700
+++ tools/module_naming_scheme/hierarchical_mns.py	2015-10-13 19:13:00.860190573 -0700
@@ -152,7 +152,7 @@
         tc_comp_info = self.det_toolchain_compilers_name_version(tc_comps)
 
         paths = []
-        if modclass == MODULECLASS_COMPILER or ec['name'] in ['CUDA']:
+        if modclass == MODULECLASS_COMPILER:
             # obtain list of compilers based on that extend $MODULEPATH in some way other than <name>/<version>
             extend_comps = []
             # exclude GCC for which <name>/<version> is used as $MODULEPATH extension
```

I'm not used to the framework code but it seems that it considers cuda as a compiler even if its easyconfig defines it as a system moduleclass, but then this code excludes GCC from extend_comps and finally request both GCC and CUDA versions, and that fails in KeyError:
```
$ eb CUDA-7.5.18.eb 
== temporary log file in case of crash /tmp/eb-V_Zxo0/easybuild-hRVPlK.log
== processing EasyBuild easyconfig /global/easyconfigs/CUDA-7.5.18.eb
== building and installing Core/CUDA/7.5.18...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== postprocessing...
== sanity checking...
ERROR: Traceback (most recent call last):
  File ".../easybuild_framework-2.3.0-py2.6.egg/easybuild/main.py", line 112, in build_and_install_software
    (ec_res['success'], app_log, err) = build_and_install_one(ec, init_env)
  File ".../easybuild/framework/easyblock.py", line 1994, in build_and_install_one
    result = app.run_all_steps(run_test_cases=run_test_cases)
  File ".../easybuild/framework/easyblock.py", line 1934, in run_all_steps
    self.run_step(step_name, step_methods)
  File ".../easybuild/framework/easyblock.py", line 1816, in run_step
    m(self)
  File ".../easybuild/framework/easyblock.py", line 1899, in <lambda>
    (SANITYCHECK_STEP, 'sanity checking', [lambda x: x.sanity_check_step()], False),
  File ".../easybuild/easyblocks/c/cuda.py", line 97, in sanity_check_step
    super(EB_CUDA, self).sanity_check_step(custom_paths=custom_paths)
  File ".../easybuild/framework/easyblock.py", line 1593, in sanity_check_step
    fake_mod_data = self.load_fake_module(purge=True)
  File ".../easybuild/framework/easyblock.py", line 1031, in load_fake_module
    fake_mod_path = self.make_module_step(fake=True)
  File ".../easybuild/framework/easyblock.py", line 1695, in make_module_step
    txt += self.make_module_extend_modpath()
  File ".../easybuild/framework/easyblock.py", line 957, in make_module_extend_modpath
    modpath_exts = ActiveMNS().det_modpath_extensions(self.cfg)
  File ".../easybuild/framework/easyconfig/easyconfig.py", line 1158, in det_modpath_extensions
    modpath_extensions = self.mns.det_modpath_extensions(self.check_ec_type(ec))
  File ".../easybuild/tools/module_naming_scheme/hierarchical_mns.py", line 176, in det_modpath_extensions
    comp_name_ver = [comp_name, comp_ver_tmpl % comp_versions]
KeyError: 'GCC'
```

As a side note however, my installation is based on 2.3.0 and not on _develop_ so sorry if I'm completely misleading you.

**Comment by boegel** [comment]
@thiell: there's a long standing open pull request that is probably related to your problem, see https://github.com/hpcugent/easybuild-framework/pull/1086

Maybe comment on that to bring it back in view.

Maybe @geimer has some feedback on this too, since he has looked into the hierarchical module name scheme extensively. But let's follow it up in https://github.com/hpcugent/easybuild-framework/pull/1086...

There's nothing wrong with this easyconfig file, the problem lies in HierarchicalMNS, i.e. in the EasyBuild framework. And it matters little whether you're using 2.3.0 or `develop` in this case, you'll run into the same problem with `develop` afaik.

**Comment by thiell** [comment]
@boegel: thx, done.

**Pull request merged**